esp-hal/src/hmac.rs

```diff
@@ -23,16 +16,23 @@
 //!   downstream mode).
 //! - Re-enables soft-disabled JTAG (in downstream mode).
 //!
-//! # Availability on ESP32 family
+//! ## Configuration
+//! The HMAC module can be used in two modes - in ”upstream” mode the HMAC
+//! message is supplied by the user and the calculation result is read back by
+//! the user. In ”downstream” mode the HMAC module is used as a Key Derivation
+//! Function (KDF) for other internal hardwares.
+//!
+//! ### HMAC padding
 //!
-//! The accelerator is available on ESP32-S2, ESP32-S3, ESP32-C3 and ESP32-C6.
+//! The HMAC padding is handled by the driver. In downstream mode, users do not
+//! need to input any message or apply padding. The HMAC module uses a default
+//! 32-byte pattern of 0x00 for re-enabling JTAG and a 32-byte pattern of 0xff
+//! for deriving the AES key for the DS module.
 //!
-//! # HMAC padding
+//! ## Examples
+//! Visit the [HMAC] example for an example of using the HMAC accelerator.
```

Comment:
Maybe we should avoid using two "example" words in a row? (just a nitpick)

Something like: "Visit the [HMAC] example to learn how to use the HMAC accelerator"